Source/OCMock/OCMockObject.m

```diff
@@ -508,4 +574,46 @@ - (NSString *)_stubDescriptions:(BOOL)onlyExpectations
 }
 
 
+@end
+
+/**
+ * The observer gets installed the first time a mock object is created (see +[OCMockObject initialize]
+ * It stops all the mocks that are still active when the testcase has finished.
+ * In many cases this should break a lot of retain loops and allow mocks to be freed.
+ * More importantly this will remove mocks that have mocked a class method and persist across testcases.
+ * It intentionally turns off the assert that fires when calling a mock after stopMocking has been
+ * called on it, because when we are doing cleanup there are cases in dealloc methods where a mock
+ * may be called. We allow the "assert off" state to persist beyond the end of -testCaseDidFinish
+ * because objects may be destroyed by the autoreleasepool that wraps the entire test and this may
+ * cause  mocks to be called. The state is global (instead of per mock) because we want to be able
+ * to catch the case where a mock is trapped by some global state (e.g. a non-mock singleton) and
+ * then that singleton is used in a later test and attempts to call a stopped mock.
+ **/
+@implementation OCMockXCTestObserver
+
+- (BOOL)conformsToProtocol:(Protocol *)aProtocol {
+    // This allows us to avoid linking XCTest into OCMock.
+    return strcmp(protocol_getName(aProtocol), "XCTestObservation") == 0;
+}
+
+- (void)testSuiteWillStart:(XCTestCase *)testCase {
+    gAssertOnCallsAfterStopMocking = YES;
+    gCurrentMocksToStopRecorder = gTestSuiteMocksToStop;
```

Comment:
If the state and logic would live in its own class, it would be possible to make the code more explicit, e.g. that class could have an explicit flag signalling which collection to use and then have an `if` statement to put the added mock to the right collection.